packages/mobile/src/store/personal-info/actions.ts

```diff
@@ -0,0 +1,35 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { SaveUserProfileDto } from '@vse-bude/shared';
+import { AsyncThunkConfig } from '~/common/types/types';
+import {
+  personalInfoParser,
+  updatePersonalInfoParser,
+} from '~/helpers/helpers';
+import { ActionType } from './common';
+
+const getPersonalInfo = createAsyncThunk<
+  SaveUserProfileDto,
+  undefined,
+  AsyncThunkConfig
+>(ActionType.GET_USER_BY_ID, async (_, { extra }) => {
+  const { personalInfoApi } = extra;
+  const response = await personalInfoApi.getPersonalInfo();
+  const personalInfo = personalInfoParser(response);
+
+  return personalInfo;
+});
+
+const updatePersonalInfo = createAsyncThunk<
+  SaveUserProfileDto,
+  SaveUserProfileDto,
+  AsyncThunkConfig
+>(ActionType.UPDATE_PERSONAL_INFO, async (payload, { extra }) => {
+  const { personalInfoApi } = extra;
+  const parsedPayload = updatePersonalInfoParser(payload);
+  const response = await personalInfoApi.updatePersonalInfo(parsedPayload);
+  const personalInfo = personalInfoParser(response);
```

Comment:
These data manipulations are not obvious. Consider removing them at all or keep is as helper and apply before dispatching the action

Reply:
corrected